src/engine/sparqlExpressions/SparqlExpressionTypes.h

```diff
@@ -313,20 +312,27 @@ std::optional<ExpressionResult> evaluateOnSpecializedFunctionsIfPossible(
   return result;
 }
 
-/// An Operation that consists of a `FunctionAndValueGetters` that takes
-/// `NumOperands` parameters. The `FunctionForSetOfIntervalsType` is a function,
-/// that can efficiently perform the operation when all the operands are
-/// `SetOfInterval`s.
-/// It is necessary to use the `FunctionAndValueGetters` struct to allow for
-/// multiple `ValueGetters` (a parameter pack, that has to appear at the end of
-/// the template declaration) and the default parameter for the
-/// `FunctionForSetOfIntervals` (which also has to appear at the end).
+// Class for an operation used in a `SparqlExpression`, consisting of the
+// function for computing the operation and the value getters for the operands.
+// The number of operands is fixed.
+//
+// NOTE: This class is defined in the namespace `sparqlExpression` and is
+// different from the class with the same name defined in `Operation.h`
+//
+// TODO: The rest of this comment seems to be outdated. Please check, Johannes.
+//
+// An Operation that consists of a `FunctionAndValueGetters` that takes
+// `NumOperands` parameters. The `FunctionForSetOfIntervalsType` is a function,
+// that can efficiently perform the operation when all the operands are
+// `SetOfInterval`s. It is necessary to use the `FunctionAndValueGetters`
+// struct to allow for multiple `ValueGetters` (a parameter pack, that has to
+// appear at the end of the template declaration) and the default parameter for
+// the `FunctionForSetOfIntervals` (which also has to appear at the end).
```

Comment:
```suggestion
// The `SpecializedFunction`s can be used to choose a more efficient implementation given the types of the operands. 
// For example, expressions like `logical-or` or `logical-and` can be implemented more efficiently if all the inputs are `SetOfInterval`s`.
```
(The mechanism has become more generic in the meantime).